src/main/preferences/schema.json

```diff
@@ -92,7 +92,10 @@
     "type": "string",
     "pattern": "^[_A-z0-9]+((-|\\s)*[_A-z0-9])*$"
   },
-
+  "trimUnnecessaryEmptyLine": {
```

Comment:
Perhaps it would be better to name this `trimUnnecessaryCodeBlockEmptyLines` to distinguish between code block and the full markdown document. I think we should add options for the markdown document: ensure single trailing newline, remove trailing newlines and smart detect the both one.

Reply:
>I think we should add options for the markdown document: ensure single trailing newline, remove trailing newlines and smart detect the both one.

What's the meaning of `both one`?

Reply:
We could also detect whether a single trailing newline exists and add or remove the new trailing newlines before saving.

Reply:
If we automatically remove the blank line at the end of the code block when editing, it may cause unable to edit, so we don't remove the blank line when saving.

Reply:
> If we automatically remove the blank line at the end of the code block when editing, it may cause unable to edit, so we don't remove the blank line when saving.

I'm talking about the full markdown document before saving the document such as vscode.

>  I think we should add options for the markdown document: ensure single trailing newline, remove trailing newlines and smart detect the both one.